core/src/ipld.rs

```diff
@@ -9,8 +9,41 @@ use alloc::{
 };
 use core::fmt;
 
-use crate::cid::Cid;
 use crate::error::TypeError;
+use crate::{cid::Cid, error::TypeErrorType};
+
+#[derive(Debug, Clone, PartialEq)]
+/// Floating point type that excludes NaN, Infinity, -Infinity, and subnormal values
```

Comment:
Out of curiosity: why forbid infinite and subnormal values? They could easily participate in an `impl Ord + Eq`.

Reply:
They are excluded by the IPLD data model if you use IEEE floats: https://github.com/ipld/ipld/blob/master/docs/data-model/kinds.md#float-kind

Reply:
Yes, but there is no further explanation why these particular values are outlawed, hence my curiosity.

Reply:
The IPLD Data Model is intentionally kept abstract. E.g. we have "integer" and "float" kinds. Within the IPLD Data Model, it's not specified how they are represented. Integers could e.g. be arbitrary precision ones in some implementations, 64 or 128-bit integers or even IEEE doubles (hello JS). The same is true for floats. I would expect that they are usually implemented as IEEE doubles. But you could also think of implementations that uses fractions instead. As we want to keep that option open, we non numeric things. There is also more information on floats at https://github.com/ipld/ipld/blob/015ff80aea78f588b981c6cef7f1e9bd59d31f48/design/tricky-choices/numeric-domain.md.